src/components/application_manager/src/resumption/resumption_data_processor.cc

```diff
@@ -68,13 +68,27 @@ void ResumptionDataProcessor::Restore(ApplicationSharedPtr application,
   SetGlobalProperties(application, saved_app);
   AddSubscriptions(application, saved_app);
   register_callbacks_[application->app_id()] = callback;
+
+  if (HasNoHMIRequestsSent(application->app_id())) {
```

Comment:
@LitvinenkoIra please rename to `IsInternalResumption` and check using saved inside that resumption does not require sending HMI requests. You can add an additional check that resumption_status_ is empty with DCHECK.